### PR TITLE
feat(HomeBrew): AND-1350 Live quotes on confirm screen

### PR DESCRIPTION
**Pull request by westonal**
Early days, but both fragments are sharing a "model", the model contains the socket.

image

### CODE REVIEW

File: morph/ui/src/main/java/com/blockchain/koin/morphUiModule.kt [hunk 1]
```diff
@@ -36,6 +37,12 @@ val morphUiModule = applicationContext {
         factory { ExchangeConfirmationPresenter() }
 
         factory { TradeHistoryPresenter(get(), get()) }
+
+        context("Quotes") {
```
Comment (westonal) [inline morph/ui/src/main/java/com/blockchain/koin/morphUiModule.kt]: Not using this yet, was thinking of clearing the context on the activities stop but with `by viewModel` I don't need to worry.

File: morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/host/HomebrewNavHostActivity.kt [hunk 2]
```diff
@@ -44,11 +52,54 @@ class HomebrewNavHostActivity : BaseAuthActivity(), HomebrewHostActivityListener
...
+    override fun launchConfirmation() {
+        navController.navigate(R.id.exchangeConfirmationFragment)
+    }
+
+    private val compositeDisposable = CompositeDisposable()
+
+    private fun newQuoteWebSocket(): QuoteService {
+        val quotesService = exchangeViewModel.quoteService
+
+        compositeDisposable += listenForConnectionErrors(quotesService)
```
Comment (westonal) [inline morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/host/HomebrewNavHostActivity.kt]: `listenForConnectionErrors` is back on the activity once more. This means it works when either fragment is on screen.

File: morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/host/HomebrewNavHostActivity.kt [hunk 3]
```diff
@@ -68,5 +119,5 @@ internal interface HomebrewHostActivityListener {
 
     fun setToolbarTitle(@StringRes title: Int)
 
-    fun launchConfirmation(confirmationModel: ExchangeConfirmationModel)
+    fun launchConfirmation()
```
Comment (westonal) [inline morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/host/HomebrewNavHostActivity.kt]: No need to pass data around. The shared android `ViewModel` means the fragments can share the socket etc.

File: morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/confirmation/ExchangeConfirmationFragment.kt [hunk 4]
```diff
@@ -88,24 +121,10 @@ class ExchangeConfirmationFragment :
...
-    val sendingCurrency: CryptoCurrency,
-    val receiving: String,
-    val receivingCurrency: CryptoCurrency,
-    val accountItem: JsonSerializableAccount
-) : Parcelable
+class ExchangeConfirmationViewModel(
+    val value: FiatValue,
+    val sending: CryptoValue,
+    val receiving: CryptoValue
+)
```
Comment (westonal) [inline morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/confirmation/ExchangeConfirmationFragment.kt]: As not serializing it anymore, I can use nice types, not primatives

File: morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/host/HomebrewNavHostActivity.kt (hunk 3 above)
Comment (ditn) [reply morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/host/HomebrewNavHostActivity.kt]: Nice.

File: morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/confirmation/ExchangeConfirmationFragment.kt (hunk 4 above)
Comment (ditn) [reply morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/confirmation/ExchangeConfirmationFragment.kt]: Yaay!

**Pull request merged**